recipes/cd3-boost-unit-definitions/all/conanfile.py

```diff
@@ -0,0 +1,121 @@
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "14.1",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        # export_conandata_patches(self)
+        pass
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        cmake_layout(self)
+
+    def requirements(self):
+        # prefer self.requires method instead of requires attribute
+        # direct dependencies of header only libs are always transitive since they are included in public headers
+        self.requires("boost/1.72.0", transitive_headers=True)
+
+    # same package ID for any package
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            # validate the minimum cpp standard supported when installing the package. For C++ projects only
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(
+            str(self.settings.compiler), False
+        )
+        if (
+            minimum_version
+            and Version(self.settings.compiler.version) < minimum_version
+        ):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        # download source package and extract to source folder
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    # not mandatory when there is no patch, but will suppress warning message about missing build() method
+    def build(self):
+        # The attribute no_copy_source should not be used when applying patches in build
+        # apply_conandata_patches(self)
+        pass
+
+    # copy all files to the package folder
+    def package(self):
+        copy(
+            self,
+            pattern="LICENSE.md",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+        )
+        copy(
+            self,
+            pattern="*.hpp",
+            dst=os.path.join(self.package_folder, "include"),
+            src=os.path.join(self.source_folder, "src"),
+        )
+
+    def package_info(self):
+        # folders not used for header-only
+        self.cpp_info.bindirs = []
+        self.cpp_info.libdirs = []
+
+        # if package provides a CMake config file (package-config.cmake or packageConfig.cmake, with package::package target, usually installed in <prefix>/lib/cmake/<package>/)
+        self.cpp_info.set_property("cmake_file_name", "BoostUnitDefinitions")
```

Comment:
👍 https://github.com/CD3/BoostUnitDefinitions/blob/0e0b14255644a1de16d7a3a02fcd00018ecabaa4/CMakeLists.txt#L57

Reply:
I'm not sure what this one means. Should I put a link to the BoostUnitDefinitions project in the recipe here as a reference for the cmake file name?

Reply:
Oh sorry, I just left a comment for my future self so I know I looked this up the first time 😉 

No action is required